### ISSUE TITLE
Open communication

### ISSUE THREAD

**Issue by ShalokShalom** [fsharp/fssf-ask-the-board#7]
This suggestion is about an improvement to the reputation for F-Sharp in the open source community by installing a social bridge to matrix.org 

# Description
[description]: #description

I plan to connect our community with people, who do not come by in our Slack, since they often refuse to use closed source software for both, technical and social issues. 

This opens the free software community for F# which then results in contributions and further adoption. :slightly_smiling_face:

# Motivation and impact
[motivation]: #motivation

Every software has its flaws, so has Slack. 
Next to the social benefit we win, there is a technical improvement as well.

There are people out there, who would never join the community, so long as its based on Slack.

By providing them a sane way to interact with the community, you give them the chance to contribute. 

There are so many people, who changed the project completely, like the developers of Ionide and Fable.

Imagine there is one single developer which becomes such a valuable contributor, simply for accepting a technical and social bridge.

The open source community is rich on talented developers, so lets connect to them. 

I meet one week ago a nice guy at a local meetup, which falls exactly in this category. :slightly_smiling_face:

I think that is a really great deal. :wink:

# Proposed solution and alternatives
[solution]: #solution

Matrix is a peer to peer communication platform, which provides a huge amount of bridges, including one to Slack.

This bridge is already implemented in their primary client, Riot.

It is also supported there, so they provide both server and the maintenance.

One other solution is, that we can connect to the IRC channel too, which connection this section of the community too.

# Drawbacks
[drawbacks]: #drawbacks

Small ones: 

People contributing from outside appear as bots, while this is quite nice implemented in this case.

There is one more instance to force the Code of Conduct, while I see luckily very less need to do so. :slightly_smiling_face:

# Are you willing to help?

Yes. 

# Other thoughts
[other]: #other-thoughts

Haskell has quite the opposite stance in the open source community, while it also gets its core contributions from  companys such as like Microsoft and Facebook. 

So why? 

.Net Core is open source, very well done and gets the same love on all supported operating systems. 

Same counts for VSCode and other development tools. 

So why is the Haskell community still significantly more considered as a serious choice for open source enthusiasts? 

Why has F# such a tricky stance in some areas of the community?

One of the main critic points which I receive for recommending F# is that "it is not really open source"

This almost always includes that Microsoft is involved, which is still considered as a non free company. 

Haskell on the other side is a very respected language in the open source scene. 

Why?

To me, solutions like the fssf are clearly huge steps into the correct direction to solve this. :slightly_smiling_face:

Slack, as an proprietary solution has a difficult stance in the community since it disrespects their data, ships with huge ressource consumption and offers not even a sane way to find old discussions. 

Some people see an opposite message sent, as for what the fssf stands:
*Isolation.*

We see now new developments such as LLVM and fez, who both are targeting on other platforms and I see a great potential future for F-Sharp as a language for different platforms, such as Kotlin and Ceylon.

In fact, it is already such one, thanks to Fable. 

This means, the community gets broader and the question about open source communication will spread.

This means, the whole question probably *stays* and even *grows*. 

Let's solve this now :wink:

:slightly_smiling_face:

**Comment by realvictorprm** [comment]
In general personally I'm in favour of using open source alternatives to Slack and will look into the one you suggested here.

However I already know that switching to a different client for the whole F# community will only be possible over years if and if only a substancial part of the community agrees on using the new communication client primarily. These days I think this is only possible if the FSSF supports the idea centrally with a makeshift you proposed here. 

Anyways, because we can't support every single movement out there this would require that there is a major existing movement using the client proving that there is much interest in the alternative.

PS: This is my opinion so far.

**Comment by realvictorprm** [comment]
Also this isn't a message like "no don't do this!". I'm rather the one who is curious and wants to try this out.

As soon as you established this chat I suggest to provide a link on this page so people can decide for themselves (Also it may be good to have a full clickable link to the message client here).

**Comment by ShalokShalom** [comment]
Matrix is the most well established solution for peer to peer communication and one of the most accepted solutions for open source communications at all.

There are around 1300 registered members in the main Rust channel and about 14000 in the Matrix channel itself.

Matrix is the new IRC. :slightly_smiling_face:

Once the FSSF officially supports that movement, is surely a lot of interest awaken on both sides. 

I can ping one guy who lets me occasionally write guest posts on his blog and I am sure we can get some announcements released on other places as well.

One more thing: I know that the community voted for Slack. The thing is, that this community is very business oriented, the already existing community voted for this. 

While this is a democratic idea does it inherently ignore some parts of the *potential* community.

I see this now as an easily done mistake in any kind of voting and I keep this in my mind.

**Comment by charlesroddie** [comment]
The OP is very rambly but this is the core:
> I plan to connect our community with people, who do not come by in our Slack, since they often refuse to use closed source software for both, technical and social issues.

This is not a serious use case. I would assume this is a tiny group of people. They also appear to be irrational and opinionated, and as a result of this toxic combination their contribution to the community could be negative.

I would very much like it if there were something better than Slack but the advantages would have to be real technical advantages (client memory usage, history...) rather than ideological ones.

**Comment by FoggyFinder** [comment]
I guess you could just create F# place on Matrix and show us how great it can be

**Comment by alanmbarr** [comment]
Could we please focus on the merit of the idea and creating an inclusive community and not focus on judging the natures of people?

**Comment by bjornbaverfjord** [comment]
The idea of reaching and communicating with as many people as possible in an efficient way is good.

It seems like a less than ideal solution to bridge two different platforms with a fragile link. If Matrix.org fits all the requirements of the fssf and there is any real benefit it would be far preferable to abandon Slack and move everything over to Matrix.org rather than having a link that will multiply the efforts of operation.

**Comment by eugene-g** [comment]
Let's add [Zulip](https://zulipchat.com/why-zulip/) to evaluation.

**Comment by mastoj** [comment]
I think we would loose more than we would win by moving to a lesser known platform than slack. You can think what you want about slack, but it is widely used and something a lot of developers already have installed on their phones and computers. So why switch to something else that forces us to install yet another application?

**Comment by EBrown8534** [comment]
I think that, assuming we want to consider such a change, we should actually gather a statistically-significant headcount of those who would benefit from such a decision (whether it be a bridge, transfer, custom client, what-have-you). Maybe a headcount of who is currently wanting to get in on the Slack discussion, but cannot (or will not) because they will not use Slack.

Regardless of the reason (ignoring that Slack is CSS and "we" want an OSS solution) we should look at _who_ will benefit first, _then_ why. Regardless of what we do, it complicates things for the board, foundation, et. al., so we should first gather an estimate of the impact.

> I meet one week ago a nice guy at a local meetup, which falls exactly in this category.

This is interesting, and I would be curious to see how many fall into this situation _and_ want to be more involved. I'm going to assume (but it's not explicitly mentioned) that this person wants to be involved in the FSSF discussion, but won't because "Slack." The keyword here is "and", we can't just go off of how many F# developers won't use Slack if they won't participate if we move to a different (or additional) medium.

I'd like to see numbers (and I'm sure others would as well). I think if they're sufficient, this would be a lot easier to discuss and/or attempt. It's not a bad idea, I just think we need to know the full scope of who's affected, as that might help determine if "Matrix.org" is the right idea, or if there is a _better_ alternative or solution.

Bottom line, I would think the next steps should be as follows:

* Gather a headcount of those wanting to replace Slack, or refusing to use Slack;
    * This could possibly be done by someone setting up discussion areas in another arena, and assessing membership;
    * Can also be done by votes on this issue (assuming those folks aren't in a mentality of not using GitHub for the same reason);
* Assess options, determine what would be most effective (I.e. is Matrix the best choice? Zulip? IRC?);
    * This is a far bigger player than it seems - folks have different preferences, and assuming that moving to Matrix would solve this is a bad assumption, because folks like me don't know what it is and don't really have the time to (or don't want to) learn;
* Submit for board approval (**we** should do the legwork before they have to, can't expect them to spend all their time on this research);

**Comment by rizary** [comment]
Hi,

I am not sure if my comment is useful or not to determine whether we need slack's alternatives or not. I am new to fsharp, but if we can move from slack to another place ( i like discord personally --i know about "I DONT WANT TO INSTALL ANOTHER SOCIAL APP" problem, sorry though) then it would be very grateful for me. One thing that is my concern is about the history. I know we can somehow archive it or use some technique, but from my pov (as beginner programmer) sometimes I have a problem with a different timezone. So when I miss 3 days or more (even a week) to open my slack to see all the chat, I cant do that because of the limited history of slack (which what I want is scroll the hell up).

and from what I have seen on another FP slack community (yeah that big one), trying to give suggestion to move from slack to another place (like discord) will end up being discouraged by several members. I have to accept the reality and use slack all the time. Maybe in fsharp slack, you have more than 2000 member who also experiences like me and tries not to speak up about changing community app. 

Another point is for an example, there is FP community in discord, but when I ask about F# channel (the place where I hope I can learn more about fsharp), they say to me that I need to join to #ocaml instead. Same as when I ask about purescript community.

So the conclusion is, when I want to learn fsharp, I will look into their official community's social apps which is right now is in slack. Since we (me personally) dont have any choice, I would use it anyway though.

**Comment by ShalokShalom** [comment]
And this is exactly the point:

Others also see themself confronted with the same situation. 

While they conclude, that the community is obviously willing to essentially give no choice.

 And this is simply considered a very unconventional practice. 

It is fine to use closed software, so long as you

A) Work on alternatives
B) Provide alternatives

This is simply how they are used to it for the whole carrier.

Nearly every single open source project hosts here on Github.

The point is:

1) Github works fine, Slack has significant issues. 

2) You can use the git commandline tool and your data is independent of their service. 

Slack forces you into what they ship and you depend on their service.

This is the difference. 

Github is much more accepted as Slack.
So, its not fanatism or religious behavior.
It is, as I said, a *social* issue.

They feel forced.
And they are potentially very helpful persons.

So its simply a pragmatic decision. 
Include them in the community. 

People feel offended by the Slack-only strategy, since they feel forced to use a tool which sucks their memory and lacks fundamental functionality. 

And how to hack that in, without the source code? 

Zulip is very nice on the desktop, while I see their client on Android is lacking behind, imho. 

Plus, it is self hosted and the foundation prefers to save money and time. 

Matrix is almost completely peer to peer based, so modern and easy to maintain, without loose of control.

Other solutions are welcome of course. ^-^

**Comment by 7sharp9** [comment]
@Rizary I like discord too, its working well for Elm and its more searchable for finding content

**Comment by eugene-g** [comment]
Sorry, I should have given more info on Zulip. The state of things is following:

- Slackarchive addresses one of the biggest issues with slack: 10k messages limitation.
- Currently there is a problem with slackarchive (@ReedCopsey is trying to resolve it)
- If slackarchive it won't be resolved, we're really need to look for alternative, since slack doesn't scale well (if number of user grows, we'll see our messages disappearing too quickly; and it applies to private messages!)
- As far as I understood, FSSF wouldn't mind paying reasonable price for good service. Unfortunately, Slack doesn't care about non-profits and their pricing is unresonable ($6.67 per user x 2k users = $160k per year)
- Zulip is totally free for non-profits. We're talking about cloud solution, not on-premises.
- It has a little bit different communication model. Maybe it addresses the issues mentioned by @Rizary and @7sharp9 

I've never tried Zulip and would just walk away(who wants to test another software?), but their friendliness to non-profits is urging me to try. So I've registered fsharp.zulipchat.com and inviting people for initial evaluation. For invitation DM me on FSSF chat, my handle is `@evgeny`.

**Comment by ShalokShalom** [comment]
I already tried Zulip and found it very appealing. When you refer to slackarchive as a solution, how different looks this solution to Matrix? 

I recommend testing the mobile clients as well and I admit, that I did the mistake to not look into their business model and pricing in regards of non profit organizations.

I also like their team, they response quickly and helpful.

**Comment by eugene-g** [comment]
> When you refer to slackarchive as a solution, how different looks this solution to Matrix?

As I stated above, Slackarchive solves the main issue with Slack, which is history limitation. It differs in being totally unobtrusive to current setup.

**Comment by ShalokShalom** [comment]
https://fsharp.slackarchive.io/

This is already implemented. 

As I pointed out, is the main concern for people like me that the source code is not open, that Slack forces me to use its ressource hungry client and that most importantly, I completely depend on them and their service. :slightly_smiling_face:

**Comment by rizary** [comment]
Ah yes.. I see someone quote my comment "**I will look into their official community's social apps**" and put in fssf to point out that there are people (including me) who cannot differentiate between FSSF and community. Maybe because English is not my first language, so he would think I'm stupid enough to see the difference. Again, being able to speak up will ended up on discouragement.

What I mean by official "community's social app" is any form of apps which I can see the update or seeing other people share their knowledge in social app or even people who just do a simple chit chat. So same as in haskell, I search in google "Fsharp Community" and "Haskell Community" everytime I want to find "Community's social app". If i can't find it, I search "Haskell Channel" or "Fsharp Channel" and found it "fsharp.org/guides/slack/", so by that looks I think FSSF encourage newbies (like me) to join there to chat and learn about fsharp.

hmm.. maybe I don't need to defend myself by saying this. Again, I'm sorry for my bad English which caused bad interpretation. The underline is I know FSSF is different from "the community" and for me, it does not matter if fsharp community is in Slack or else. I join any fsharp social app to learn about fsharp.

I think my comment is already out of topic. I will leave the discussion.

Thank you.

**Comment by ShalokShalom** [comment]
You are welcome here and I agree with you. :slightly_smiling_face:

There is now a F-Sharp chat on Matrix: 
https://matrix.to/#/#fsharp:matrix.org

**Comment by dustinmoris** [comment]
Personally I think the communication tool should be determined by its features and the barriers of getting it installed on a system. Slack is very feature rich and it runs everywhere and it is free, so the barrier is zero. If someone doesn't like Slack because it's closed source then let it be. This is not a rationale reason to not use a chat tool. There's people in this world which you can never please and I'd hate to switch to a less feature rich or crappy tool because some dude trolled about Slack being closed source. Who cares about that seriously... it's a bloody chat tool, it's not critical software. If Slack dies tomorrow then everyone can move to another equivalent chat tool in a minute, so why does it matter if it is open or closed source? I get the open source argument for pieces of software which are core parts of another commercial product where a company has a financial motivation to have some control or access to it's source code, but please don't make it a silly argument for every piece of software that is used for secondary or tertiary tasks. More rationale and perspective is needed in the software community. Slack is fine IMHO.

**Comment by ShalokShalom** [comment]
If you read this thread carefully, will you recognize that I  never mention that anybody should be forced to leave Slack, while you suggest that I do so. 

I mention clearly that there are a couple of benefits for us all, when we open our community a broader audience group and they persist, even if you decide to call them silly. 

These kind of people are responsible for our free internet today and for a lot of packages you use every day. 

Slack lacks core features like performance effectiveness, open development and a privacy to them.

If you are fine with that, is this completely fine to me, nobody here forces you. 

The thing is, that some others do care. 

You can surely explain, what are the features on either Zulip, or Matrix or other platforms you see lacking? 

And why you are in a position to declare the choices of others as unimportant and silly?

Why do you insult people simply because they put their ideas on the internet? 

In this case is the tool really irrelevant, since anti-social behavior is platform independent. 

Atom is available, and also other editors with formidable F# support.

Are you fine when the developers of VSCode decide to hide the source code, sell your privacy and somebody forces you into it?

To sum it up: You consider Slack as not critical, to me is every software critical. 

You are free to not care. 
Let me care about my point of view.

**Comment by isaacabraham** [comment]
OK, can everyone calm down :-) FWIW I'm personally not fussed about this, but that my opinion. I am interested by @ShalokShalom 's comment regarding VS Code - do you really believe that this will happen? And if so - why couldn't that happen in, say, Atom?

**Comment by ShalokShalom** [comment]
Of course does it not happen.

The question is, how he feels about it, if it would happen.

Of course he would be disappointed.

So, open source tools are great so long as they provide him as a person a direct benefit and once they are useful for somebody else, is it 'silly' and pointless. 

This is the typical attitude of somebody who interpret  open source as something purely egocentric instead as the social foundation which it is. :slightly_smiling_face:

Since open source software is always much more flexible and trustworthy due its very nature, is it obvious that such software is the preferred choice, when you compare two functionally identical options.

This is something so much commonly accepted in the scene, that a harsh violation of this core concept shows immediately, that this company/group has not yet understood the core values of the huge community. 

RedHat understands that. IBM understands that.
Millions of people who interact daily with open source software understand that. 

And I spend here hours to defense such common knowledge, which means, for everybody outside, that you are clearly not ready (as a community) for broader acceptance in the open source field. 

The numbers show that clearly: Go into a random channel and ask, how accepted Clojure, Haskell or any other language is which has a strong background of a huge Company.

You will almost never find some people, who consider Haskell less of an open source project as any other language or project. :)

While, if you do that with F#, you might get quickly the respond that there is still a strong bond to Microsoft and its platform. 

There is surely already a great deal of change done, while there is also still a lot of stuff to do.

I see a sixth step in this [imho formidable article of Krzysztof](http://kcieslak.io/Path-to-community-based-OSS)

Honor the freedom of choice. :slightly_smiling_face:

**Comment by dustinmoris** [comment]
> Are you fine when the developers of VSCode decide to hide the source code, sell your privacy and somebody forces you into it?

First of all nobody forces anyone into anything. Nobody forces anyone to use Slack either. I don't care much whether VSCode is open source or closed. I used VS for many years and never cared about it either. I am not attached to it and therefore I don't really care. Also nothing in this world is free. If a product is free as in money because it keeps its source closed so it can sell an enterprise version to paying customers, or when it uses data for ads or something else then I don't have a problem with it as long as it is transparent about these things and give me a fair choice of whether I'm ok with it or not. There's nothing wrong with trying to make money somehow from intellectual property, even a hobbyist who writes a lot of code for free has somewhere else an income source which therefore by definition must be something similar to a paid version, closed source or some other commercialised product. If everything would be entirely free and open source then we would have nothing or only shit.

I support good software by allowing creators the freedom of making a fair profit from their work. If closed source is sometimes required for that (due to the nature of the business, ease of copycats, whatever) then why should I oppose it when I am still thankful for getting a great tool for free?

I think we need to be careful that we don't make the false assumption that if we switch from Slack to "insert-open-source-chat-tool-here" that it will attract more developers than it will detract happy Slack users.

EDIT:

Do the same people who refuse to use any closed source software also refuse to live in a house where they don't have the blueprints of it, or refuse to drink coffee where they haven't seen from where each individual bean has been sourced, or refuse to eat a burger where they haven't seen which cow it came from and how the cow has been fed? Do you refuse to board a plane before seeing all the manufacturing and maintenance plans, speaking to the pilot and knowing everything about him inside out? Do you refuse to take the bus to work because the engine in the bus is proprietary technology? If not, why so short sighted/intolerant when it comes to software?

**Comment by ShalokShalom** [comment]
As said multiple times: There is no support next to Slack, so this *effectively* forces everyone who like to be part of the community into this one tool.

The ignorance about this is, as said, part of the issue.

Especially since I spelled it out several times now.  

I prefer open source and yes, I always choose that path when it comes to other stuff.

Just because its not always available (yet) does this mean to you its less senseful? 

And in a world full of open source would money be simply superfluous.

**Comment by dustinmoris** [comment]
> As said multiple times: There is no support next to Slack, so this effectively forces everyone who like to be part of the community into this one tool.

I mean right now I feel forced to use GitHub if I want to participate in this discussion. How about that? I am also forced into using Git if I want to contribute to F#. I am forced into buying a computer chip which is not open source. What annoys me the most is that I am forced to pay for an internet connection in order to participate in this community. I was forced to use Twitter to even find out about this GitHub issue. Why am I even forced to speak English right now? There's quite A LOT of things which we are all "forced" in order to participate in social communities. That's life.

But anyhow, let's forget that. I think what matters is that we should establish if we make a false assumption or not before changing chat tools:

> I think we need to be careful that we don't make the false assumption that if we switch from Slack to "insert-open-source-chat-tool-here" that it will attract more developers than it will detract happy Slack users.

**Comment by Krzysztof-Cieslak** [comment]
If anything this statement is ignorance: 

> As said multiple times: There is no support next to Slack, so this effectively forces everyone who like to be part of the community into this one tool.

F# Community is active in many different places  - GitHub, Twitter, StackOverflow, Reddit (lol), real life meetups and conferences, and probably other places I don’t know about. Saying that this particular Slack is only communication medium, and anyone is forced to use it is simply not true. Also, the Slack we discuss is *Foundation* Slack, not *”Everyone in Community”* Slack - you can participate in the F# Community not having anything to do with Foundation. F# world would be super sad if everyone would be in one place controlled by one entity 

> And in world full of open source ...

Also, I love this support to idea of OSS, that’s rich from someone who hasn’t been working everyday for last couple of years to build F# ecosystem like some people in the Community.

**Comment by ShalokShalom** [comment]
@dustinmoris 

> I mean right now I feel forced to use Github

I appreciate that feedback. Its for the people who decided to use Github for issues like that.

We can tell us stories about how sad all is, while I simply make a suggestion to use the alternatives which are available. 

I agree that there are different prerequisites to join in this discussion and software development at all and this issue is all about solving such cases. 

I never said that I think its fine that you or others are forced in some way, so why do you suggest that I do so?

I actually change this. 

> I think we need to be careful that we don't make the false assumption that if we switch from Slack to "insert-open-source-chat-tool-here" that it will attract more developers than it will detract happy Slack users.

I am sure you appreciate that I state now for the approximately forth time, that I vote for a continuation of the Slack channel. 

And I also made a couple of times clear, that there are a lot of potential developers who appreciate a significant step towards open source adoption from fsharp's side. 

While, I hope they dont find this thread.... 

@Krzysztof 

Luckily, you have no write access to this repo, otherwise would I be probably blocked from it, same as from your Elm repo. :slightly_smiling_face:

You might feel attacked, since you are going to release a closed source software, so I can ensure you that I wish you much luck with it. 

> Also, I love this support to idea of OSS, that’s rich from someone who hasn’t been working everyday for last couple of years to build F# ecosystem like some people in the Community.


You obviously contributed a lot to the community, which is imho zero vindication to put someone the ass into his face, call him religious fanatic, ignore the messages of the board leaders and suggest that someone has no right to spread its word, simply because he has not enough contributed to the community in your eyes. 

Since this behavior is pure poison to every community. 

> F# Community is active in many different places - GitHub, Twitter, StackOverflow, Reddit (lol), real life meetups and conferences, and probably other places I don’t know about.

So go and tell me how many competent people are there. Of course it it possible to get some kind of support on these places, so my statement might be a bit bold. 

The thing is that the community circles around the Slack channel, the majority of people are there and it is a live chat, compared to all the other solutions.

I consider this as important, while those other solutions are of course also part of the community. 

Again: This is no attempt to shut down the Slack channel, it is the idea of expanding the community.

**Comment by ghost** [comment]
I came to this thread via [F# Weekly](https://sergeytihon.com/category/f-weekly/). 
I think this whole discussion has way too much priority and even I am forced (@dustinmoris 🤣 , I had a good laugh) to participate now, because this thread is given a priority through F# Weekly it really should not have.

First of all I love the development of OSS in recent years. I love all the dedication and passion that goes into it from developers all around the world and that everybody despite his background comes to a common understanding of concepts and things through working on the same code. That is beautiful to me, an expression of freedom and one of the greatest things that happen in our crazy world.

But IMO there is another factor so a OSS product can become really great and that is **commercial backing**. 
Just think about Docker (Moby) or Kubernetes. Hell I would so not use them with confidence and build my career around those technologies if it wasn't also for the billions of $ that go into making them secure and performant and all the upstream technologies around them. Just my hypothesis but without all that I bet they would be an inaccessible and unusable piece of shit.

I think @Krzysztof-Cieslak and @dustinmoris (and many others) would be even more productive and would bring their projects to an even better state if there are no more existential risks to them in any way due to solid and reliable cash flow for what they love to do and atm probably spend their free time for.  

And imagine there would be companies with a real commercial interest and stakes in the quality of the products they maintain. More flaws per unit time get removed and everything around those projects like marketing and documentation would also step up, just because of the dead simple demand for them. Those things are so important for getting more adoption and resources, however there are no resources yet to fire them up properly. It's virtually rocket science 😄. And all work and effort should go into getting the rocket into space now. And we just need some money to fire up the process now.

I've been wanting  to go all-in F# since I first encountered it. and the fact that there simply is no real job market yet even kind of drove me into self-employment and OSS contribution. What an egoistic motive that is 😄. 

And the SAFE-Stack is the best thing that could possibly happen to F#. That's why I decided to contribute now is just because I am enthusiastic about it and truly believe again that mainstreaming F# can still work.  This whole discussion here just rises the opposite feeling in me.  

Who gives a F# about Slack or Matrix. There is a time and a place for things. And now is the time to become attractive to companies with a lot of cash behind them.  So they in turn create F# jobs for people who want to enter a SAFE happy place with maximum native code sharing where they use all the great stuff the F# community has created so far. They will love it and spread the word. Other companies then want it. More jobs, more developers, more word-spreading. More sold licenses for Neptune and more projects that need consultancy. I am sure @Krzysztof-Cieslak would like it. And it is ok to like that and to want that without giving up idealism.

The best thing I can imagine right now to do for F# is something I never dared or specifically wanted to do : just spread the word about all the great things. This is why I got involved with the MeetUp in Zurich and I will give a first talk in Summer and I try to make it dazzling for an .NET audience. 

The title is a bit bold for my first public talk ever: **"The Tale of FSharp: A Language For The Next Decade: Act I : RAD Reloaded"**. I am currently working on it and try to fiddle out the best way to organize code to take the SAFE-stack to the next level by also incorporating Xamarin.Forms and WPF in the code sharing scenario with Elmish. The older .NET senior developers still secretly love VB6 and RAD (Rapid Application Development) with its UI first approach. So lets give them their beloved RAD back, but on steroids with F# and Elmish. I don't know if that will make any significant difference for F# but I know for sure that it will as least not do as much damage as a GitHub issue facing the public that reveals such severe personal differences within the community. 

If F# Weekly puts something like this in my face and mind, I had to do a little personal promo at least 😄 . 

@ShalokShalom 
I am quite sure @Krzysztof-Cieslak was merely getting emotional over some personal stuff you 2 have going on and did not mean to offend or attack anyone who did not contribute to F# as much as he did.
My post is 90% off-topic but so is this whole thing anyway. And I really like consistency 😄

**Comment by ShalokShalom** [comment]
Your post is completely on topic, imho. ^-^

I think you see open source as something in contrast to money?

Ask RedHat, a multi billion dollar company.
Ask Linux, LibreOffice, Firefox, Gnome, KDE.

Ask openSUSE and ask Microsoft in a few years, if they see the chances who lay behind 'sharing source'

And ask all the others, who raise millions of dollars and users by respecting the open source principles. 

And by the way: I have no clue, how the optional adoption of another communication platform stops you from your plan anyway?

How does it hinder you to promote F# to companies?

I think that makes it quite clear: 
You see open source as anything to the contrary to the business world?

If so, does this show clearly that you look with a very naive view on the topic, since quite the opposite is the case. :slightly_smiling_face:

Show me one company, which avoids open source software? Show me one company, which is turned of by it. They may release their own software closed, while their own stack of software is almost always significantly open.

So why do you suggest that opening a Matrix channel can in any sense hinder you to make promotion in the big shops?

You could be not more incorrect, as by saying this has too much priority:

This is one of the keys to the healthy growth, which you like to achieve, imho. 

I still consider F# as one of the most sophisticated technologies around here, while the blind ignorance and complete lack of understanding of open source shows me *clearly* that this is a major drawback to everyone, who sees how the software industry is working.

You want on the Linux servers?
This is clearly, how to avoid them. 

Company's often refuse to adopt F# since a) the community is small and b) the primary audience group which is targeted is the .Net world:

C# developers are pretty fine with their language. 

They are used to it and they ignore the fact that fsharp is pretty much a superset of their baby. 
Yes, you can show them a couple of amazing benefits in F#. 

And then they realize that it means the adoption of a complete new concept. They most likely develop ideas to avoid this new language, since they can fail and this is then obviously based on scare, which is a very strong power. 

So, its tricky to adopt them. 

To switch on a similar language is tricky enough and to switch on a complete new concept is very scary, at least in their book. 

So, how to change that?

You can say, think and feel about the open source enthusiasts what you want:

They potentially bring you this boost.
The same counts for newbies:

Complete programming newbies can show impressive results within a few years. Since F# provides immutability, conciseness and type safety together with sane tooling and compiler messages. And an enormous ecosystem. And hey: Also a high amount of pretty competent people to support those complete beginners.

F# can be the new Python. 
And you can see that professionals will recognize this. 

Which will result in higher adoption.
Since hey: 'If they can do it, I can do so as well.'

And again: The community shows so less competence here, that I am simply wondering, if thats a joke?

I consider this as solved.

**Comment by 7sharp9** [comment]
This has to be the best Ironically posted discussion on communication
skills.

**Comment by realvictorprm** [comment]
@7sharp9 let's keep on topic

**Comment by ghost** [comment]
I bet the number of comment edits is a very good metric on how emotional the people writing them are in the moment of writing.  Also the amount of text in a response in combination with the initial response time could be a supporting factor. Is that what they mean when they say "All we need to know about people is in the metadata"? 😃 

Why not conclude this whole thing by fetching the publicly available data of this thread and analysing it. There could be some features that act as good predictors for communication in public threads going into a toxic direction. Let's think about Communication itself before thinking about the Communication tool of choice 😄 .

**Comment by ShalokShalom** [comment]
> There could be some features that act as good predictors for communication in public threads going into a toxic direction. 

Ignorance.

**Comment by realvictorprm** [comment]
@kfrie I appreciate your comment because clearly _ignorance_ as supposed by @ShalokShalom isn't the solution.

**Comment by ShalokShalom** [comment]
I did not intend to present it as a solution, while I guess you mean `described ` instead of `supposed`? 

I simply see it as an indicator. If one ignores the other, this will not lead into productive communication. 

You can see several times here, that this suggestion is assumed as a thread on the Slack channel. 
I repeatedly argued, that this is about open communications and providing other people access to the community. 

When something goes such clearly misunderstood, once again and again, anyway if you repeat your intention and the plan to realize it, is this clearly on purpose. 

This is a thread on the plans of others and by that a so called [Projection](https://en.wikipedia.org/wiki/Psychological_projection)

This shows also up by using other defense mechanisms, like [argumentum ad hominem](https://en.wikipedia.org/wiki/Ad_hominem)

**Comment by ghost** [comment]
Also never forget the virtue of ignorance. Just imagine that magnificent, utopian world, where people can read and write stuff in the internet without being and acting offended with each other and just chill a little if they do not like something. Just imagine how wonderful that world would be ...

Although that already would be some form of projection technically. Does that already match the pathological definition and was does pathological mean anyways? Is traditional psychology pseudo-scientific and Freud established his own Oedipus-Complex as a general rule as part of his pathological projections (lol) ? Or are they real sciences and they just store all their theorems in mutable variables? A clear indicator towards this hypothesis is the fact, that gay people before 1987 where appearently suffering under a sexual orientation disturbance, whereas people after 1987 were not, [according to DSM](https://www.psychologytoday.com/us/blog/hide-and-seek/201509/when-homosexuality-stopped-being-mental-disorder).

**Comment by ShalokShalom** [comment]
Well, the Wikipedia article is very clear about this. Besides this: Look how mental illnesses got 'cured' before Freud and Co appeared: They treated people with electric shocks and lobotomy.

**Comment by baronfel** [comment]
Everyone,

Thank you for the spirited discussion on this ticket.  Many opinions have been expressed and I think we're all aware of the arguments now.  The Board will discuss this ticket at the next meeting (approx. 1 week) and will respond at some time afterwards.  In the meantime, due to the somewhat wide-ranging discussion, I'm going to lock this issue from further discussion until we have a response.

Thank you all.

**Comment by ReedCopsey** [comment]
The Board of Trustees discussed this, in the context of a larger discussion about Slack, at today's Board meeting. I have posted a new issue (#8), and encourage people to provide feedback there.

Thank you!
